src/App.tsx

```diff
@@ -1,26 +1,214 @@
+      .then(newTodo => {
+        setTodos(currentTodo => [...currentTodo, newTodo]);
+      })
+      .catch(() => {
+        setError(ErrorMessage.AddTodos);
+
+        throw new Error();
+      })
+      .finally(() => {
+        setTempTodo(null);
+        setIsInput(false);
+
+        returnFocus();
+      });
+  };
+
+  const handleDeleteTodo = (todoId: number) => {
+    setProcessingTodos(currentIds => [...currentIds, todoId]);
+
+    return todoService
+      .deleteTodos(todoId)
+      .then(() =>
+        setTodos(currentTodos =>
+          currentTodos.filter(todo => todo.id !== todoId),
+        ),
+      )
+      .catch(() => setError(ErrorMessage.DeleteTodos))
+      .finally(() => {
+        setProcessingTodos(currentIds =>
+          currentIds.filter(id => id !== todoId),
+        );
+
+        returnFocus();
+      });
+  };
+
+  const handleClearCompleted = () => {
+    [...todos]
+      .filter(todo => todo.completed)
+      .forEach(todo => {
+        handleDeleteTodo(todo.id);
+      });
+  };
+
+  const handleSetError = (newError: ErrorMessage) => {
+    setError(newError);
+  };
+
+  const handleRemoveError = () => {
+    setError(null);
+  };
+
+  useEffect(() => {
+    if (!todoService.USER_ID) {
+      return;
+    }
+
+    setError(null);
+    field.current?.focus();
+
+    todoService
+      .getTodos()
+      .then(setTodos)
+      .catch(() => setError(ErrorMessage.LoadTodos));
+  }, []);
+
+  useEffect(() => {
+    if (!error) {
+      return;
+    }
+
+    const timer = setTimeout(() => {
+      setError(null);
+    }, 3000);
+
+    return () => clearTimeout(timer);
+  }, [error]);
+
+  if (!todoService.USER_ID) {
     return <UserWarning />;
   }
 
   return (
-    <section className="section container">
-      <p className="title is-4">
-        Copy all you need from the prev task:
-        <br />
-        <a href="https://github.com/mate-academy/react_todo-app-add-and-delete#react-todo-app-add-and-delete">
-          React Todo App - Add and Delete
-        </a>
-      </p>
-
-      <p className="subtitle">Styles are already copied</p>
-    </section>
+    <div className="todoapp">
+      <h1 className="todoapp__title">todos</h1>
+      <div className="todoapp__content">
+        <TodoHeader
+          onAdd={handleAddTodo}
+          onError={handleSetError}
+          onFocus={field}
+          isInput={isInput}
+          isComleted={allTodoIsCompleted}
+          changeAllComplete={handleChangeAllComplete}
+          todos={todos}
+        />
+        <TodoList
+          processingTodos={processingTodos}
+          todos={todos}
+          filterValue={filtering}
+          onDelete={handleDeleteTodo}
+          tempTodo={tempTodo}
+          todoChanger={handleUpdateTodo}
+        />
+        {todos.length !== 0 && (
+          <TodoFooter
+            todos={todos}
+            filterValue={filtering}
+            setFilter={(value: FilterType) => {
+              setFiltering(value);
+            }}
```

Comment:
```suggestion
            setFilter={setFiltering}
```
You still use a function wrapper, but inline. Just pass the `setFiltering` function